src/ScheduleAddWindow.cpp

```diff
@@ -38,46 +39,60 @@ enum {
-	BLayoutBuilder::Group<>(this, B_VERTICAL, 0).SetInsets(0).Add(back).End();
-	back->SetViewUIColor(B_PANEL_BACKGROUND_COLOR);
-
-	BString label;
-	label.SetToFormat(B_TRANSLATE("Type: %s"), data.Type().Type());
-	BStringView* typelabel = new BStringView("typelabel", label.String());
-
-	label.SetToFormat(B_TRANSLATE("Payee: %s"), data.Payee());
-	BStringView* payeelabel = new BStringView("payeelabel", label.String());
-
+#undef B_TRANSLATION_CONTEXT
+#define B_TRANSLATION_CONTEXT "SplitView"
+
+	// Type
+	BStringView* typeLabel = new BStringView("typelabel",B_TRANSLATE("Type"));
+	typeLabel->SetExplicitSize(BSize(be_plain_font->StringWidth("ShortType"), B_SIZE_UNSET));
+	BTextControl* type = new BTextControl("type", NULL, data.Type().Type(), NULL);
+	type->SetEnabled(false);
+
+	// Payee
+	BStringView* payeeLabel = new BStringView("payeelabel",B_TRANSLATE("Payee"));
+	payeeLabel->SetExplicitMaxSize(BSize(B_SIZE_UNLIMITED, B_SIZE_UNSET));
+	payeeLabel->SetExplicitMinSize(BSize(be_plain_font->StringWidth("anAveragelyLongPayee"),
+		B_SIZE_UNSET));	BTextControl* payee = new BTextControl("payee", NULL, data.Payee(), NULL);
+	payee->SetEnabled(false);
+
+	// Amount
+	BStringView* amountLabel = new BStringView("amountlabel",B_TRANSLATE("Amount"));
+	amountLabel->SetExplicitSize(BSize(be_plain_font->StringWidth("$10,000,000,000.00"),
+		B_SIZE_UNSET));
 	BString temp;
 	gCurrentLocale.CurrencyToString(data.Amount().AbsoluteValue(), temp);
-	label.SetToFormat(B_TRANSLATE("Amount: %s"), temp.String());
-	BStringView* amountlabel = new BStringView("amountlabel", label.String());
+	BTextControl* amount = new BTextControl("amount", NULL, temp.String(), NULL);
+	amount->SetEnabled(false);
 
-	label = B_TRANSLATE("Category");
-	label << ": ";
+	// Category
+	BStringView* categoryLabel = new BStringView("categorylabel",B_TRANSLATE("Category"));
+	categoryLabel->SetExplicitSize(BSize(be_plain_font->StringWidth("aLongCategoryName"),
+		B_SIZE_UNSET));
+	BString label;
 	if (data.CountCategories() > 1)
 		label << B_TRANSLATE("Split");
 	else
 		label << data.NameAt(0);
+	BTextControl* category = new BTextControl("category", NULL, label.String(), NULL);
+	category->SetEnabled(false);
 
-	BStringView* categorylabel = new BStringView("categorylabel", label.String());
-
-	label = B_TRANSLATE("Memo");
-	label << ": " << data.Memo();
-	BStringView* memolabel = new BStringView("memolabel", label.String());
+	// Memo
+	BStringView* memoLabel = new BStringView("memolabel",B_TRANSLATE("Memo"));
+	memoLabel->SetExplicitMaxSize(BSize(B_SIZE_UNLIMITED, B_SIZE_UNSET));
+	memoLabel->SetExplicitMinSize(BSize(be_plain_font->StringWidth("$10,000,000,000.00"),
+		B_SIZE_UNSET));
+	BTextControl* memo = new BTextControl("memo", NULL, data.Memo(), NULL);
+	memo->SetEnabled(false);
 
-	//	Since layout-api, we need other way to make divider
-	//	BBox *divider = new BBox(r);
-	//	AddChild(divider);
+#undef B_TRANSLATION_CONTEXT
+#define B_TRANSLATION_CONTEXT "ScheduleAddWindow"
```

Comment:
Shouldn't this be at the top of the file?

Reply:
In this case I added a `#define B_TRANSLATION_CONTEXT "SplitView" in line 48 to have that context for all strings inbetween. This should avoid translation work.
Though I think, since this code is peppered with identical strings - Amount, Category, Date etc. everywhere - I might create a new `B_TRANSLATION_CONTEXT "CommonStrings" ` for those later. Unfortunately, that means re-generating the catkeys and having Polyglot invalidate them again...
`

Reply:
No problem then.